client_plugin/drivers/shared/kvstore/etcdops/etcdops.go

```diff
@@ -657,3 +668,132 @@ func (e *EtcdKVS) DeleteMetaData(name string) error {
 	}
 	return nil
 }
+
+// AtomicIncr - Increase a key value by 1
+func (e *EtcdKVS) AtomicIncr(key string) error {
+	// Create a client to talk to etcd
+	etcdAPI := e.createEtcdClient()
+	if etcdAPI == nil {
+		return fmt.Errorf(etcdClientCreateError)
+	}
+	defer etcdAPI.Close()
+
+	ticker := time.NewTicker(checkSleepDuration)
+	defer ticker.Stop()
+	timer := time.NewTimer(requestTimeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+			resp, err := etcdAPI.Get(ctx, key)
+			cancel()
+			if err != nil {
+				log.WithFields(
+					log.Fields{"key": key,
+						"error": err},
+				).Error("Failed to Get key-value from ETCD Client ")
+				return err
+			}
+
+			oldVal := string(resp.Kvs[0].Value)
+			num, _ := strconv.Atoi(oldVal)
+			num++
+			newVal := strconv.Itoa(num)
+			if e.CompareAndPut(key, oldVal, newVal) {
+				return nil
+			}
+		case <-timer.C:
+			return fmt.Errorf("Timeout reached; AtomicIncr is not complete")
+		}
+	}
+}
+
+// AtomicDecr - Decrease a key value by 1
+func (e *EtcdKVS) AtomicDecr(key string) error {
+	// Create a client to talk to etcd
+	etcdAPI := e.createEtcdClient()
+	if etcdAPI == nil {
+		return fmt.Errorf(etcdClientCreateError)
+	}
+	defer etcdAPI.Close()
+
+	ticker := time.NewTicker(checkSleepDuration)
+	defer ticker.Stop()
+	timer := time.NewTimer(requestTimeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
+			resp, err := etcdAPI.Get(ctx, key)
+			cancel()
+			if err != nil {
+				log.WithFields(
+					log.Fields{"key": key,
+						"error": err},
+				).Error("Failed to Get key-value from ETCD Client ")
+				return err
+			}
+
+			oldVal := string(resp.Kvs[0].Value)
+			num, _ := strconv.Atoi(oldVal)
+			if num == 0 {
+				return fmt.Errorf("Cannot decrease a value equal to 0")
+			}
+			num--
+			newVal := strconv.Itoa(num)
+			if e.CompareAndPut(key, oldVal, newVal) {
+				return nil
+			}
+		case <-timer.C:
+			return fmt.Errorf("Timeout reached; AtomicDecr is not complete")
+		}
+	}
+}
+
+// BlockingWaitAndGet - Blocking wait until a key value becomes equal to a specific value
+// then read the value of another key
+func (e *EtcdKVS) BlockingWaitAndGet(key string, value string, newKey string) (string, error) {
+	// Create a client to talk to etcd
+	etcdAPI := e.createEtcdClient()
+	if etcdAPI == nil {
+		return "", fmt.Errorf(etcdClientCreateError)
+	}
+	defer etcdAPI.Close()
+
+	ticker := time.NewTicker(checkSleepDuration)
+	defer ticker.Stop()
+	timer := time.NewTimer(2 * requestTimeout)
```

Comment:
This operation may need to wait remote SMB server starts. So it can take up to 5 to 10 seconds. So we doubled the requestTimeout, which is 5 seconds.
All the timeout currently are just estimation. We need more testing to get a more accurate timeout value.